zerver/actions/message_delete.py

```diff
@@ -153,3 +163,70 @@ def do_delete_messages_by_sender(user: UserProfile) -> None:
     )
     if message_ids:
         move_messages_to_archive(message_ids, chunk_size=retention.STREAM_MESSAGE_BATCH_SIZE)
+
+
+def delete_deactivated_user_messages(
+    realm: Realm,
+    user_profile: UserProfile,
+    message_delete_action: MessageDeleteAction,
+    acting_user: UserProfile | None,
+) -> None:
+    delete_public_stream_messages = message_delete_action.get(
+        "delete_public_stream_messages", False
+    )
+    delete_private_stream_messages = message_delete_action.get(
+        "delete_private_stream_messages", False
+    )
+    delete_direct_messages = message_delete_action.get("delete_direct_messages", False)
```

Comment:
I think we wouldn't need this code if we defined `message_delete_action` as a Pydantic dataclass with default values of `false` for each of these fields.

Additionally, I think we could call `message_delete_action` something like `deactivate_user_actions: DeactivateUserActions`